dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/dataContainer.py

```diff
@@ -1947,18 +1946,17 @@ def get_body_for_evalmode(self):
                 text_to_return.append("    if (modeChangeType == NO_MODE)\n")
                 text_to_return.append("      modeChangeType = DIFFERENTIAL_MODE;\n")
             else:
-                print "Mode not handled"
+                print("Mode not handled")
             text_to_return.append("  }\n")
             text_to_return.append("\n")
         for z in self.modes_discretes:
             discrete_mode = self.modes_discretes[z]
             for eq in discrete_mode.eqs:
                 text_to_return.append("  // ----- Mode for " + str(eq) + " --------- \n")
-            zAff = to_param_address(z)
-            zPre = zAff.replace("localData[0]->discreteVars", "simulationInfo->discreteVarsPre")
-            zPre = zPre.replace("localData[0]->integerDoubleVars", "simulationInfo->integerDoubleVarsPre")
-            text_to_return.append("  if (" + zAff + " != " + zPre +")\n")
-            text_to_return.append("  {\n")
+            z_aff = to_param_address(z)
```

Comment:
As you are doing corrections, it would be good to add a line testing the address: test_param_adress(z) before using the to_param_address method.